docs/api/java/getting-started.md

```diff
@@ -422,402 +437,536 @@ public class Main {
-                fee.getErgsPriceLimitNumber(),
-                signer.getAddress(),
-                meta
-        );
+    BigInteger amount = Token.ETH.toBigInteger(0.5);
 
-        String signature = signer.getDomain().thenCompose(domain -> signer.signTypedData(domain, transaction)).join();
-        byte[] message = TransactionEncoder.encode(transaction, TransactionEncoder.getSignatureData(signature));
+    // ETH By default
+    TransactionReceipt receipt = wallet.withdraw("0x<receiver_address>", amount).send();
 
-        String sentTransactionHash = zksync.ethSendRawTransaction(Numeric.toHexString(message)).send().getTransactionHash();
+    // Also we can withdraw ERC20 token
+    Token token;
 
-        // You can check the transaction status
-        TransactionReceipt receipt = zksync.ethGetTransactionReceipt(sentTransactionHash).send().getTransactionReceipt();
-    }
+    // ERC20 token
+    TransactionReceipt receipt = wallet.withdraw("0x<receiver_address>", amount, token).send();
+  }
 }
-
 ```
 
-### Withdraw funds via ZkSyncWallet
+### Finalize Withdraw
 
 ```java
-
-import io.zksync.protocol.core.Token;
+import io.zksync.protocol.ZkSync;
+import io.zksync.protocol.provider.EthereumProvider;
+import org.web3j.crypto.Credentials;
+import org.web3j.protocol.Web3j;
 import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import org.web3j.tx.RawTransactionManager;
+import org.web3j.tx.TransactionManager;
+import org.web3j.tx.gas.ContractGasProvider;
+import org.web3j.tx.gas.StaticGasProvider;
 
 import java.math.BigInteger;
 
 public class Main {
-    public static void main(String... args) {
-        ZkSyncWallet wallet; // Initialize wallet
-
-        BigInteger amount = Token.ETH.toBigInteger(0.5);
-
-        // ETH By default
-        TransactionReceipt receipt = wallet.withdraw("0x<receiver_address>", amount).send();
-
-        // Also we can withdraw ERC20 token
-        Token token;
-
-        TransactionReceipt receipt = wallet.withdraw("0x<receiver_address>", amount, token).send();
-    }
+  public static void main(String... args) {
+    Web3j web3j; //Initialize web3j
+    ZkSync zksync; //Initialize zksync
+    Credentials credentials; //Initialize credentials
+    BigInteger chainId; //Initialize chainId
+
+    TransactionManager manager = new RawTransactionManager(web3j, credentials, chainId.longValue());
+    BigInteger gasPrice = web3j.ethGasPrice().send().getGasPrice();
+    ContractGasProvider gasProvider = new StaticGasProvider(gasPrice, BigInteger.valueOf(300_000L));
+    TransactionReceipt receipt = EthereumProvider
+      .load(zksync, web3j, manager, gasProvider).join()
+      .finalizeWithdraw("TRANSACTION_HASH", 0);
+  }
 }
-
 ```
 
 ## Wallet
 
-Get the price of the execution of the transaction
+## Contract interaction using wrapper
 
-```java
+### Fee
 
-import io.zksync.crypto.signer.EthSigner;
+#### Get price of the transaction execution (currently not wokring properly)
```

Comment:
we have 3 titles here. If this script and the next one are related to Fees, let's just group them under an h3 "Calculate transaction fees"

Reply:
Maybe it will be a bit more noticeable like this :
> **Warning:** Feature currently unsupported. Under development